File: solid/Customer.java
package solid;

public class Customer implements UserManagement, SubscriptionNotificationService {

    private String name;
    private String email;

    public Customer(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    @Override
    public void subscribeToNewProductsAvailability() {
        System.out.println("Subscribing to new products availability...");
    }

    @Override
    public void subscribeToSMSNotifications() {
        System.out.println("Subscribing to new products availability...");
    }

    @Override
    public void updateUserProfile(Customer customer) {
        System.out.println("Updating user profile + " + customer.getName() + "...");
    }

    @Override
    public void changePassword(Customer customer, String newPassword) {
        System.out.println("Updating user: " + customer.getName() + "...");
    }
}

File: typescript/solid-principles/Customer.ts
import { SubscriptionNotificationService } from "./subscription-notification-service";
import { UserManagement } from "./user-management";

export class Customer implements UserManagement, SubscriptionNotificationService  {
    private name: string;
    private email: string;

    constructor(name: string, email: string){
        this.name = name;
        this.email = email;
    }

    getName(): string {
        return this.name;
    }

    getEmail(): string {
        return this.email;
    }

    subscribeToNewProductsAvailability(): void {
        throw new Error("Method not implemented.");
    }

    subscribeToSMSNotifications(): void {
        throw new Error("Method not implemented.");
    }

    updateUserProfile(customer: Customer): void {
        console.log("Updating user profile + " + customer.getName() + "...");
    }

    changePassword(customer: Customer, newPassword: string) {
        console.log("Updating user: " + customer.getName() + "...");
    }


}